Tests/AsyncAlgorithmsTests/TestShare.swift

```diff
@@ -572,6 +572,35 @@ final class TestShare: XCTestCase {
     XCTAssertEqual(results1, [1, 2, 3, 4, 5])
     XCTAssertEqual(results2, [])  // Should be empty since source is exhausted
   }
+  
+  func test_share_rethrows_failure_type_on_backported() async {
+    let shared = AsyncThrowingStream<Void, Error> {
+      $0.finish(throwing: TestError.failure)
+    }.share()
+    do {
+      for try await _ in shared {
+        XCTFail("Expected to not get here")
+      }
+    } catch {
+      XCTAssertEqual(error as? TestError, .failure)
+    }
+  }
+  
+  func test_share_rethrows_failure_type_without_falling_back_to_any_error() async {
+    if #available(AsyncAlgorithms 1.2, *) {
+      // Ensure - at compile time - that error is effectively a TestError
+      let shared: some AsyncSequence<Void, TestError> = FailingSequence(TestError.failure).share()
+      do {
+        for try await _ in shared {
+          XCTFail("Expected to not get here")
+        }
+      } catch {
+        XCTAssertEqual(error, TestError.failure)
+      }
+    } else {
+      throw XCTSkip("This test is not available before 1.2")
```

Comment:
I don't think this can actually hit right?

Reply:
The issue is that XCTest does not support availability macro directly on the function due to how tests are automatically "expanded" from the suite. This means that running this test on older platforms would result in an empty one. 
It felt more appropriate to expose it as a skipped test, but it's not a big deal to remove it